alchemiscale/storage/statestore.py

```diff
@@ -1199,6 +1304,14 @@ def create_task(
 
         """
         scope = transformation.scope
+        if "Transformation" not in transformation.qualname:
+            raise ValueError(
+                "`transformation` ScopedKey does not correspond to a `Transformation`"
+            )
+
+        if extends is not None and extends.qualname != "Task":
```

Comment:
useful check, I'm sure there are other sharp edges of this nature floating around.

Reply:
There most certainly are; I tripped on this one myself while doing live testing. I'll sweep through the other methods that are susceptible to this problem and add appropriate checks.

Reply:
I think I've filled many of the similar potholes in.